jax/api.py

```diff
@@ -2074,3 +2074,49 @@ def fun_invertible(*args, **kwargs):
                                    concrete=concrete)
     return tree_unflatten(out_tree(), out_flat)
   return fun_invertible
+
+def make_hlo(fun: Callable,
```

Comment:
Why not extend `xla_computation` with more arguments instead of wrapping it?

Reply:
well at the moment because people just asked I submit this one ;)  - the other issue is that the properties available on the compiled modules last I checked differed quite a lot from the uncompiled ones, so it's not so easy to offer a unified top-level api to both.  I'm not sure what's missing in the bindings underneath to get e.g. the dot-graph for the compiled version, but last time I checked you couldn't fetch such things...

Reply:
> the other issue is that the properties available on the compiled modules last I checked differed quite a lot from the uncompiled ones

Sorry I'm a bit confused, don't they both return strings?

E.g. why not add a bool `optimize` argument to `xla_computation`?

Reply:
No, `xla_computation` returns a Python-wrapped `XlaComputation` which has the following API:
https://github.com/tensorflow/tensorflow/blob/1bc83bae4f28f7576465f0d23a2ad9a700796198/tensorflow/compiler/xla/python/xla.cc#L736

The compiled and uncompiled objects are (naturally) different because one is an `Executable`, the other is an `XlaComputation`.

Note there are more things you can do with an `XlaComputation`, e.g. compile it, serialize it to a protobuf, build a dot graph, etc.

Reply:
It would definitely be nice to be able to get a dot graph from the compiled HLO the way the profiler does!